CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.12.4)
```

Comment:
This is quite a big jump, and for example goes beyond the Ubuntu 18.04 default: https://packages.ubuntu.com/bionic/cmake

Which features require this version?

Reply:
Agreed that it makes sense to restrict the CMake version requirement to popular long term Linux distros. Ubuntu 16.04 is the oldest one with available package data and that uses CMake 3.5.1, released in 2016. Will put up a patch that uses 3.5.1 as the minimum. Nothing in the change to relies on newer features than this.

Reply:
Lowered to 3.5.1 in the latest patchset, please review.

Reply:
Yes, 3.5.1 is fine with me, thanks!